### ISSUE TITLE
Flutter Web Throws Error "Expected a value of type 'String', but got one of type '_Future<String>'"

### ISSUE THREAD

**Issue by agdsat134** [zino-app/graphql-flutter#687]
**Describe the bug**
Getting the following error when compiling Star Wars example for Flutter Web.
```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following TypeErrorImpl was thrown building MyApp(dirty):
Expected a value of type 'String', but got one of type '_Future<String>'

  MyApp file:///F:/graphql-flutter/examples/starwars/lib/main.dart:21:23

When the exception was thrown, this was the stack:
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/eC:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 62:3        castError
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 442:10  cast
ypes.dart 412:9        as
packages/graphql/src/cache/normalized_in_memory.dart 21:8
new
packages/graphql/src/cache/optimistic.dart 66:8
new
packages/graphql_flutter/src/caches.dart 33:8
new
packages/starwars_app/client_provider.dart 15:31
get cache
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 816:6   get
packages/starwars_app/client_provider.dart 38:14
clientFor
packages/starwars_app/client_provider.dart 51:17
new
packages/starwars_app/main.dart 26:12
build
packages/flutter/src/widgets/framework.dart 4585:28
build
packages/flutter/src/widgets/framework.dart 4511:15
performRebuild
packages/flutter/src/widgets/framework.dart 4227:5
rebuild
packages/flutter/src/widgets/framework.dart 4490:5
[_firstBuild]
packages/flutter/src/widgets/framework.dart 4485:5
mount
packages/flutter/src/widgets/framework.dart 3455:13
inflateWidget
packages/flutter/src/widgets/framework.dart 3223:18
updateChild
packages/flutter/src/widgets/binding.dart 1132:16
[_rebuild]
packages/flutter/src/widgets/binding.dart 1103:5
mount
packages/flutter/src/widgets/binding.dart 1045:16
<fn>
packages/flutter/src/widgets/framework.dart 2612:19
buildScope
packages/flutter/src/widgets/binding.dart 1044:12
attachToRenderTree
packages/flutter/src/widgets/binding.dart 924:24
attachRootWidget
packages/flutter/src/widgets/binding.dart 906:7
<fn>
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart 48:19           internalCallback
```

**To Reproduce**
1. Remove "Platform" call from top of main.dart (because that's not support in web yet; replace with "String host = ''10.0.2.2'"
2. Compile Star Wars example with command "flutter run -d chrome"

**Web (please complete the following information):**
```
Flutter is already up to date on channel beta
Flutter 1.19.0-4.3.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 8fe7655ed2 (2 weeks ago) • 2020-07-01 14:31:18 -0700
Engine • revision 9a28c3bcf4
Tools • Dart 2.9.0 (build 2.9.0-14.1.beta)
```

**Comment by jerryzhoujw** [comment]
I got this same error when I use flutter beta to debug web.

**Comment by micimize** [comment]
haven't gotten around to this but it's possible this was fixed by `4.0.0-beta.1`

**Comment by micimize** [comment]
Closed by #754
